http/lbtransport/policy.go

```diff
@@ -108,48 +110,52 @@ func (rr *roundRobinPolicy) blacklistTarget(target *Target) {
 	rr.blacklistedTargets[*target] = rr.timeNow()
 }
 
-func (rr *roundRobinPolicy) isTrialDialingDisabled() bool {
+func (rr *roundRobinPolicy) isBlacklistDisabled() bool {
 	return rr.blacklistBackoffDuration == (0 * time.Microsecond)
 }
 
-func (rr *roundRobinPolicy) Pick(r *http.Request, currentTargets []*Target) (*Target, error) {
+func (rr *roundRobinPolicy) Picker() LBPolicyPicker {
+	return &roundRobinPolicyPicker{
+		base:           rr,
+		localBlacklist: make(map[Target]struct{}),
+	}
+}
+
+type roundRobinPolicyPicker struct {
+	base *roundRobinPolicy
+
+	// To make sure we only retry for valid targets.
+	localBlacklist map[Target]struct{}
```

Comment:
To make sure we end up our `retry` loop inside lbtransport (: even when there is no global blacklist